lib/class-wp-rest-image-editor-controller.php

```diff
@@ -101,90 +98,157 @@ public function permission_callback( $request ) {
 	public function apply_edits( $request ) {
 		require_once ABSPATH . 'wp-admin/includes/image.php';
 
-		$params = $request->get_params();
+		$params        = $request->get_params();
+		$attachment_id = $params['media_id'];
 
-		$media_id = $params['media_id'];
+		// This also confirms the attachment is an image.
+		$image_file = wp_get_original_image_path( $attachment_id );
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
 
-		// Get image information.
-		$attachment_info = wp_get_attachment_metadata( $media_id );
-		$media_url       = wp_get_attachment_image_url( $media_id, 'original' );
+		if ( ! $image_meta || ! $image_file ) {
+			$error = __( 'Unable to get meta information for file.', 'gutenberg' );
+			return new WP_Error( 'rest_unknown_attachment', $error, array( 'status' => 404 ) );
+		}
+
+		// Check if we need to do anything.
+		$rotate = 0;
+		$crop   = false;
 
-		if ( ! $attachment_info || ! $media_url ) {
-			return new WP_Error( 'rest_unknown_attachment', __( 'Unable to get meta information for file.', 'gutenberg' ), array( 'status' => 404 ) );
+		if ( ! empty( $params['rotation'] ) ) {
+			// Rotation direction: clockwise vs. counter clockwise.
+			$rotate = 0 - (int) $params['rotation'];
 		}
 
-		$meta = array( 'original_name' => basename( $media_url ) );
+		if ( isset( $params['x'], $params['y'], $params['width'], $params['height'] ) ) {
+			// Check if the crop dimensions are whitin bounds.
+			if (
+				( $params['x'] >= 0 && $params['x'] < 100 ) &&
+				( $params['y'] >= 0 && $params['y'] < 100 ) &&
+				( $params['width'] >= 1 && $params['width'] <= 100 ) &&
+				( $params['height'] >= 1 && $params['height'] <= 100 )
```

Comment:
We should put these bounds checks in the parameter args definition. We can use `maximum` to set a max value. You may need to use `exclusiveMaximum` for the x and y.

Reply:
Sounds good. Was just a bit unsure if floats would work, but with `width` and `height` expecting minimum `1`, `x` and `y` can be set to expect maximum `99` (inclusive). (Cannot have an image cropped to 1% width but starting point be over 99% of the width, etc.)

Reply:
Thinking a bit more, the API should only care if these numbers are percents, 0 - 100. The rest should be handled when doing the cropping.